# Cleveland Department of Health Complaints Analysis
An analysis of reports made to the Cleveland department of health based on data pulled from the OpenData CLE portal

## Setup

In [47]:
import os

import pandas as pd
from dotenv import load_dotenv
from sqlalchemy import create_engine

In [48]:
load_dotenv()
engine = create_engine(
    f"postgresql+psycopg://{os.environ['PG_USER']}:{os.environ['PG_PASSWORD']}"
    f"@localhost:5432/{os.environ['PG_DB']}"
)

## Data Quality Log

| Column/Issue | What I found | How I handled it | Details |
| ------------ | ------------ | ---------------- | ------- |
| `mac_complaint_id` | Mostly null, mixed formats (date-seq, truncated, six-digit) | Excluded from analysis | Too sparse to support any conclusion |
| `complaint_status` | One row with `NULL` status | Excluded via `IS NOT NULL` operator | [see Question 1](#question-1-most-frequent-complaint-types) |
| `submit_datetime` | Portal docs give no timezone, format only | Parsed and stored as simple `timestamp` | [see Question 2](#question-2-longest-open-complaints-by-type-5-complaints-per-type-threshold) |


## Question 1: Most frequent complaint types

**Data quality note:** `complaint_status` contains one row with a `NULL` value. Utilizing `IS NOT NULL` makes sure to only return rows that have a valid complaint status attached.

In [58]:
query = """
SELECT complaint_type, COUNT(complaint_type) as number_of_complaints
FROM complaints_raw
WHERE complaint_type IS NOT NULL
GROUP BY complaint_type
ORDER BY number_of_complaints DESC
LIMIT 10;
"""

df = pd.read_sql_query(query, con=engine)

print(df)

              complaint_type  number_of_complaints
0    Accumulation of Garbage                 14244
1           High Grass/Weeds                  7494
2           Rodent Abatement                  3958
3            Food Complaints                  3473
4                       Mold                  2564
5  Insect/Vermin Infestation                  2218
6            Animal Nuisance                  1370
7                      Other                  1270
8      Unsanitary Conditions                  1176
9           Health Condition                   736


**Findings:** Over all years on record, "Accumulation of Garbage" is the most reported complaint (14244 complaints), almost double the second place complaint of "High Grass/Weeds" (7494 complaints). 

## Question 2: Longest-open complaints by type (5+ complaints per type threshold)

**Data quality note:** `submit_datetime` has no documented timezone (city portal
only specifies the format, not the zone). Values appear consistent with EST
daylight hours but this is guessed, not confirmed. Stored as a `timestamp`
(no offset) rather than asserting a timezone the data doesn't explicitly mention.

In [ ]:
# For complaint types with more than 5 currently open complaints, what is the age of the
# single, longest open complaint for each type?
# TODO: finish Q2 query
query = """
WITH active_complaints AS (
    SELECT 
)

SELECT * FROM active_complaints;
"""

df = pd.read_sql_query(query, con=engine)

print(df)

<!-- TODO: write up findings for Q2 -->
**Findings:**

## Summary
<!-- TODO: write up project summary -->